src/main/java/io/cryostat/agent/FlightRecorderModule.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent;
+
+import java.lang.management.ManagementFactory;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import jdk.jfr.FlightRecorder;
+import jdk.jfr.Recording;
+import jdk.management.jfr.ConfigurationInfo;
+import jdk.management.jfr.FlightRecorderMXBean;
+
+public class FlightRecorderModule {
+
+    private FlightRecorderMXBean bean;
+
+    public FlightRecorderModule() {
+        bean = ManagementFactory.getPlatformMXBean(FlightRecorderMXBean.class);
+    }
+
+    public void startRecording(String templateName) {
+        if (isValidTemplate(templateName)) {
+            long recordingId = bean.getRecordings().size() + 1;
+            bean.startRecording(recordingId);
+            bean.setConfiguration(recordingId, templateName);
+        }
+    }
+
+    public boolean isValidTemplate(String templateName) {
+        for (ConfigurationInfo config : bean.getConfigurations()) {
+            if (config.getName().equals(templateName)) {
```

Comment:
Maybe it would be nice to allow matching on both the filename and the Label? https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management.jfr/jdk/management/jfr/ConfigurationInfo.html#getLabel() ex. `default.jfc` is also called `Continuous`.

Reply:
I think this might be nice to support since in the Cryostat UI, it's that Label that is displayed and used for selecting event templates, rather than the actual filename. So this might just reduce a little friction and make things feel similar across both components of the product.